### ISSUE TITLE
setTimeout(x) value doesn't work intermittently

### ISSUE THREAD

**Issue by srathor8x8** [fracpete/rsync4j#45]
Hi - We are using this library to run some rsync operations and have used a output processing timeout to be 1800 secs(30 mins). Once in a while, it not working and the thread which runs the rsync process gets stuck at at wait() method here in monitor() (AbstractProcessOutput.java).

<img width="819" alt="image" src="https://github.com/fracpete/rsync4j/assets/102510512/80a84ec3-13a5-4a13-ab74-c2f0c9822eb2">
 This thread got stuck for several days and was recovered by manual intervention. 
We are using rsync version rsync4j-all:3.2.3-10.

**Comment by fracpete** [comment]
I've pushed out a new release of processoutput4j which you could try instead of the version that rsync4j uses:

```xml
    <dependency>
      <groupId>com.github.fracpete</groupId>
      <artifactId>processoutput4j</artifactId>
      <version>0.0.12</version>
    </dependency>
```

**Comment by srathor8x8** [comment]
Does it contain any such probable fixes?

**Comment by fracpete** [comment]
Yes, I made the release specifically for that purpose.

**Comment by srathor8x8** [comment]
Okay Thanks. So basically, if we are using rsync-3.3.0-0, internally it shall already be using 0.0.12 for processoutput4j, right?

**Comment by fracpete** [comment]
You can also use your current rsync4j version (there is no rsync4j release with processoutput4j 0.0.12 yet), simply upgrade the processoutput4j dependency.

**Comment by srathor8x8** [comment]
Ok. Thanks. Few questions here:
1. As we have the new rsync lib also available, what would be merit of using this version over 3.2.3-10. 
2. And if use it - are there any kind of regressions which are possible and we should possible test?
3. Similarly for processoutpu4j new version, what kind of area we should focus more on testing considering this is a upgraded new release.

**Comment by fracpete** [comment]
1. See the release notes of [rsync4j](https://github.com/fracpete/rsync4j/releases) and the release notes of [rsync](https://download.samba.org/pub/rsync/NEWS#3.3.0) itself (Windows users only due to bundling of binaries, Linux/Mac check your underlying OS binary of rsync for version)
2. See 1.
3. processoutput4j - the only change was in regards to timeout handling, very minimal impact.

**Comment by fracpete** [comment]
due to no further feedback, assuming to be fixed (96e3db70c0f8b9c3225dac75d9aa637e8423a103)
